gnovm/pkg/gnolang/go2gno.go

```diff
@@ -549,22 +567,39 @@ func (g *gnoImporter) ImportFrom(path, _ string, _ types.ImportMode) (*types.Pac
 }
 
 func (g *gnoImporter) parseCheckMemPackage(mpkg *std.MemPackage) (*types.Package, error) {
+	// This map is used to allow for function re-definitions, which are allowed
+	// in Gno (testing context) but not in Go.
+	// This map links each function identifier with a closure to remove its
+	// associated declaration.
+	var delFunc map[string]func()
+	if g.allowRedefinitions {
+		delFunc = make(map[string]func())
+	}
+
 	fset := token.NewFileSet()
 	files := make([]*ast.File, 0, len(mpkg.Files))
 	var errs error
 	for _, file := range mpkg.Files {
+		// Ignore non-gno files.
+		// TODO: support filetest type checking. (should probably handle as each its
+		// own separate pkg, which should also be typechecked)
 		if !strings.HasSuffix(file.Name, ".gno") ||
-			endsWith(file.Name, []string{"_test.gno", "_filetest.gno"}) {
-			continue // skip spurious file.
+			strings.HasSuffix(file.Name, "_test.gno") ||
+			strings.HasSuffix(file.Name, "_filetest.gno") {
```

Comment:
Why do we have some special treatment for `*_filetest.gno`? Could these files just not be `*_test.gno`?

Reply:
:smiling_face_with_tear: 

Unfortunately, we have filetests and we have them in `examples` code; and as an official, documented feature of `gno test`. (see `gno test -h`). We have [previously discussed this at the Rouen retreat](https://github.com/gnolang/gno/issues/1346). These are of the same format of the tests we have in `gnovm/tests/files`, similar to the ones you originally created for Yaegi. And I agree; they should probably not exist for anyone who isn't a language developer.

Alas, it's out of scope of this PR to tackle this; so we have to ignore those as well.